monitoring/mock_uss/interuss_logging/routes_interuss_log.py

```
@@ -0,0 +1,83 @@
+    SCOPE_SCD_QUALIFIER_INJECT,
+)
+from monitoring.monitorlib.mock_uss_interface.interuss_interaction import (
+    Interaction,
+    Issue,
+)
+
+from monitoring.mock_uss.interuss_logging.config import KEY_LOG_DIR
+
+
+@webapp.route("/mock_uss/interuss/log", methods=["GET"])
+@requires_scope([SCOPE_SCD_QUALIFIER_INJECT])
+def interuss_log() -> Tuple[str, int]:
+    """
+    Returns all the interaction logs with requests that were
+    received or initiated between 'from_time' and now
+    Eg - http:/.../mock_uss/interuss/log?from_time=2023-08-30T20:48:21.900000Z
+    """
+    from_time_param = request.args.get("from_time")
+    from_time = StringBasedDateTime(from_time_param)
+    log_path = webapp.config[KEY_LOG_DIR]
+    n = len(os.listdir(log_path))
+
+    if not os.path.exists(log_path):
+        abort(404)
+    objs = []
+    for fname in os.listdir(log_path):
+        with open(os.path.join(log_path, fname), "r") as f:
+            obj = {}
+            try:
+                for obj in yaml.full_load_all(f):
+                    interaction = ImplicitDict.parse(obj, Interaction)
+                    if (
+                        ("received_at" in interaction.query.request)
+                        and interaction.query.request.received_at.datetime
+                        > from_time.datetime
+                    ):
+                        objs.append(obj)
+                    elif (
+                        "initiated_at" in interaction.query.request
+                        and interaction.query.request.initiated_at.datetime
+                        > from_time.datetime
+                    ):
+                        objs.append(obj)
+                    else:
+                        logger.debug(
+                            f"There is no received_at or initiated_at request after {from_time_param} the request in {fname}"
+                        )
+            except (ConstructorError, KeyError) as e:
```

This silently swallows errors from the client's perspective so they may not realize that something is wrong.  We should fail 500 if this ever occurs to make sure the problem is surfaced quickly and obviously.

---

I was not being sure. The reason was that then we ll loose the information about the interactions that did happen with the required info. And we would loose them as we will delete them as part of teardown of the test cases or test scenarios. So thought of logging as an error.
Do you think it will be ok to lose that info?

---

Same comment as [above](https://github.com/interuss/monitoring/pull/186/files/42da8ec0575d9f3272d0f9925b5f9b4de2e24e97#r1313370265) -- we want to always be in a known state or else fail safe.  If we don't know what would cause these errors, we should fail safe whenever they happen.  If we know what will cause them and we know it's ok to continue in those circumstances, only then is it ok to continue.